zvariant/src/type.rs

```diff
@@ -140,6 +143,26 @@ where
     }
 }
 
+/// A helper type to serialize a tuple whose elements implement [DynamicType] but not [Type].
+///
+/// This is required because tuples already have an implementation of [DynamicType] via the blanket
+/// implementation of [DynamicType] where `T: Type`, but that results in a bound of [Type] on each
+/// element, which is stronger than needed for serializing.
+#[derive(Debug, Copy, Clone)]
+pub struct SerializeTuple<T>(pub T);
+
+impl Type for SerializeTuple<()> {
+    fn signature() -> Signature<'static> {
+        Signature::from_static_str_unchecked("")
+    }
+}
+
+impl<T: Serialize> Serialize for SerializeTuple<T> {
```

Comment:
I know it's not needed for #542 but would be good to add `Deserialize` impl as well, while you're at it.